lntest/itest/lnd_routing_test.go

```diff
@@ -1150,12 +1150,53 @@ func testInvoiceRoutingHints(net *lntest.NetworkHarness, t *harnessTest) {
 	// Now that the channels are open, we'll take down Eve's node.
 	shutdownAndAssert(net, t, eve)
 
+	// We'll need the short channel ID of the channel between Alice and Bob
+	// to make sure the routing hint is for this channel.
+	listReq := &lnrpc.ListChannelsRequest{}
+	ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+	listResp, err := net.Alice.ListChannels(ctxt, listReq)
+	require.NoError(t.t, err, "unable to retrieve alice's channels")
+
+	var aliceBobChanID uint64
+	for _, channel := range listResp.Channels {
+		if channel.RemotePubkey == net.Bob.PubKeyStr {
+			aliceBobChanID = channel.ChanId
+		}
+	}
+
+	require.NotZero(t.t, aliceBobChanID,
+		"channel between alice and bob not found")
+
 	// Create an invoice for Alice that will populate the routing hints.
 	invoice := &lnrpc.Invoice{
 		Memo:    "routing hints",
 		Value:   int64(chanAmt / 4),
 		Private: true,
 	}
+	checkInvoiceHints(ctxb, t.t, net, aliceBobChanID, invoice)
+
+	// Now that we've confirmed the routing hints were added correctly, we
+	// can close all the channels and shut down all the nodes created.
+	closeChannelAndAssert(t, net, net.Alice, chanPointBob, false)
+	closeChannelAndAssert(t, net, net.Alice, chanPointCarol, false)
+	closeChannelAndAssert(t, net, net.Bob, chanPointBobCarol, false)
+	closeChannelAndAssert(t, net, net.Alice, chanPointDave, false)
+
+	// The channel between Alice and Eve should be force closed since Eve
+	// is offline.
+	closeChannelAndAssert(t, net, net.Alice, chanPointEve, true)
+
+	// Cleanup by mining the force close and sweep transaction.
+	cleanupForceClose(t, net, net.Alice, chanPointEve)
+}
+
+// checkInvoiceHints verifies that the provided invoice results in the expected
+// hints.
+func checkInvoiceHints(ctxb context.Context, t *testing.T,
```

Comment:
nit: maybe this can be made into a closure inside the test since it's not used anywhere else?